### PR TITLE
feat: TD-23978 tmq get assignment and seek offset

### PR DESCRIPTION
**Pull request by AdamEECS**

### CODE REVIEW

File: taos-sys/src/tmq/ffi.rs [hunk 1]
```diff
@@ -137,6 +149,43 @@ extern "C" {
     pub fn tmq_get_vgroup_id(res: *mut TAOS_RES) -> i32;
 }
 
+#[cfg(taos_tmq_offset_seek)]
+extern "C" {
+    pub fn tmq_get_topic_assignment(
+        tmq: *mut tmq_t,
+        pTopicName: *const c_char,
```
Comment (zitsen) [inline taos-sys/src/tmq/ffi.rs:156]: use snake-case variable names

File: taos-sys/build.rs [hunk 2]
```diff
@@ -22,6 +22,9 @@ fn version2features<V: Into<Version>>(version: V) -> Vec<&'static str> {
         if version >= Version::new(3, 0, 2, 0) {
             feats.push("req_id");
         }
+        if version >= Version::new(3, 0, 4, 1) {
+            feats.push("tmq_offset_seek");
```
Comment (zitsen) [inline taos-sys/build.rs:26]: Is it merged into main for version 3.0.4.2?

**Review by zitsen — CHANGES_REQUESTED** [review]

File: taos-optin/src/tmq/raw.rs [hunk 3]
```diff
@@ -38,6 +39,26 @@ pub(super) mod tmq {
...
+                .to_string()
+            }
+        }
+
+        pub fn subscription(&self) -> Topics {
+            let tl = Topics::new(self.tmq.list_api);
+
+            unsafe { (self.tmq.tmq_subscription)(self.as_ptr(), &mut tl.as_ptr()) }
+                .ok_or("get topic list failed")
+                .expect("get topic shoudl always success");
```
[the remark this answers is not in the document]
Comment (AdamEECS) [reply taos-optin/src/tmq/raw.rs]: Ok, also fixed in taos-sys/src/tmq/raw.rs

File: taos-sys/build.rs (hunk 2 above)
Comment (AdamEECS) [reply taos-sys/build.rs:26]: It will merged for version 3.0.5.0

**Review by zitsen — APPROVED** [review]

**Pull request merged**